Prose/ProseLib/Tests/ProseCoreTCATests/TypesEncodingTests.swift

```diff
@@ -0,0 +1,106 @@
+//
+// This file is part of prose-app-macos.
+// Copyright (c) 2022 Prose Foundation
+//
+
+import ProseCoreTCA
+import XCTest
+
+final class TypesEncodingTests: XCTestCase {
+  let encoder: JSONEncoder = {
+    let encoder = JSONEncoder()
+    // Sort object keys so results are predictable
+    encoder.outputFormatting = .sortedKeys
+    return encoder
+  }()
+
+  func check<T: Encodable>(_ value: T, equals expected: String) throws {
+    let data = try self.encoder.encode(value)
+    let string = String(data: data, encoding: .utf8).expect("\(value) has invalid UTF-8")
+    XCTAssertEqual(string, expected)
+  }
+
+  func check<T: Encodable>(_ value: T, in expectedOneOf: [String]) throws {
+    let data = try self.encoder.encode(value)
+    let string = String(data: data, encoding: .utf8).expect("\(value) has invalid UTF-8")
+    XCTAssertTrue(expectedOneOf.contains(string))
+  }
+
+  func testJIDEncoding() throws {
+    func test(_ jid: JID, equals expected: String) throws {
+      try self.check(jid, equals: expected)
+    }
+
+    try test("test@prose.org", equals: #""test@prose.org""#)
+    try test("test@prose.org/resource", equals: #""test@prose.org""#)
```

Comment:
Try to keep the `XCTAssert…` as close as possible to the actual test. If it's not possible at least pass #file and #line through to it. Otherwise test failures are not shown in the right position - they are, just where you wouldn't expect them.

In this case you could return the JSON string from your `check` method and do the XCTAssertEqual inline without really losing anything.

Reply:
I will do that